services/forgotPassword.js

```diff
@@ -0,0 +1,36 @@
+const jwt = require("jsonwebtoken");
+const User = require("../db/models/User");
+const sendMailService = require("./sendemail");
+const userRepository = require("../repositories/user");
+const env = require("../configuration/env")
+const { generateToken } = require("../utils/helper");
+
+async function setResetPasswordToken(user) {
+  const token = generateToken({ userId: user._id }, "1h");
+  return token;
+}
+
+async function forgotPassword(email) {
+  const user = await userRepository.getUserByEmail(email);
+
+  if (!user) {
+    throw new Error(errors.EMAIL_NOT_FOUND);
+  }
+
+  const token = await setResetPasswordToken(user);
```

Comment:
we don't need extra function for this. We already have generateToken function use it here